src/test/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationonsameline/InputAnnotationOnSameLineCheck.java

```diff
@@ -0,0 +1,26 @@
+package com.puppycrawl.tools.checkstyle.checks.annotation.annotationonsameline;
+
+public class InputAnnotationOnSameLineCheck {
+
+    @Annotation int x;
+
+    int y;
+
+    @Annotation
+    @SomeClass.Annotation
+    public int getX() {
```

Comment:
added cases from issue and for all of tokens except of DOT (i can't find any usage of annotation with this token)